Explorer/Assets/Scripts/CrdtEcsBridge/JsModulesImplementation/SDKObservableEventsEngineAPIImplementation.cs

```diff
@@ -0,0 +1,44 @@
+using CRDT.Deserializer;
+using CRDT.Protocol;
+using CRDT.Protocol.Factory;
+using CRDT.Serializer;
+using CrdtEcsBridge.OutgoingMessages;
+using CrdtEcsBridge.PoolsProviders;
+using CrdtEcsBridge.UpdateGate;
+using CrdtEcsBridge.WorldSynchronizer;
+using DCL.ECS7;
+using SceneRunner.Scene.ExceptionsHandling;
+using SceneRuntime.Apis.Modules.EngineApi.SDKObservableEvents;
+using System.Collections.Generic;
+using Utility.Multithreading;
+
+namespace CrdtEcsBridge.JsModulesImplementation
+{
+    public class SDKObservableEventsEngineAPIImplementation : EngineAPIImplementation, ISDKObservableEventsEngineApi
+    {
+        public HashSet<ProcessedCRDTMessage> OutgoingCRDTMessages { get; } = new ();
+
+        public SDKObservableEventsEngineAPIImplementation(ISharedPoolsProvider poolsProvider, IInstancePoolsProvider instancePoolsProvider, ICRDTProtocol crdtProtocol, ICRDTDeserializer crdtDeserializer, ICRDTSerializer crdtSerializer, ICRDTWorldSynchronizer crdtWorldSynchronizer, IOutgoingCRDTMessagesProvider outgoingCrtdMessagesProvider, ISystemGroupsUpdateGate systemGroupsUpdateGate, ISceneExceptionsHandler exceptionsHandler, MutexSync mutexSync) : base(poolsProvider, instancePoolsProvider, crdtProtocol, crdtDeserializer, crdtSerializer, crdtWorldSynchronizer, outgoingCrtdMessagesProvider, systemGroupsUpdateGate, exceptionsHandler, mutexSync)
+        {
+        }
+
+        protected override void SyncCRDTMessage(ProcessedCRDTMessage message)
+        {
+            OutgoingCRDTMessages.Add(message);
```

Comment:
I think you should filter out messages based on their type before adding to the collection. It does not make sense to store all of them (and later iterate over them) as you are interested only in a fraction.

Reply:
Uhmm, true, I'll filter based on components then, and not type because I need all the CRDT message types: PUT, APPEND and DELETE.

Reply:
Added filter in [this commit](https://github.com/decentraland/unity-explorer/pull/662/commits/1e3524645884966c175d837160c6ecd2216a2f49) + also the `EnableSDKObservableMessagesDetection` flag to only enable all this detection logic when the scene actually subscribes to any observable and avoid that otherwise